sentry_streams/sentry_streams/pipeline/message.py

```diff
@@ -14,11 +14,14 @@
     cast,
 )
 
-from sentry_streams.rust_streams import PyAnyMessage, RawMessage
+from sentry_streams.rust_streams import PyAnyMessage, PyWatermark, RawMessage
 
 TPayload = TypeVar("TPayload")
 
-RustMessage = Union[PyAnyMessage, RawMessage]
+# Represents the Rust message types which can be sent into a delegate step
+RustMessage = Union[PyAnyMessage, PyWatermark, RawMessage]
+# Represents the Rust message types which are actually processed by delegate steps
+KafkaMessage = Union[PyAnyMessage, RawMessage]
```

Comment:
I'd recommend against this separation.  There are two issues.
- RustMessage represented the message that we can move back and forth between Rust and Python, it does not necessarily represent a KafkaMessage, so we should not rename it to KafkaMessage.
- If you need to extend RustMessage adding another type you should have a Union of a Union and the new type.

```
RustApplicationMessage = Union[PyAnyMessage, RawMessage]
RustMessage = Union[RustApplicationMessage, RawMessage]
```